initrd-trustee_plus/scripts/aa-client-service.sh

```diff
@@ -0,0 +1,154 @@
+#!/bin/bash
+. /lib/dracut-lib.sh
+set -x
+
+# Configuration
+KBS_URL="http://10.72.140.122:31968"
+ROOT_DEVICE="/dev/vda4"
+CRYPT_ROOT_NAME="luks-root"
+BOOT_DEVICE="/dev/disk/by-label/boot"
+BOOT_MOUNT="/mnt/boot_partition"
+MARKER_FILE="${BOOT_MOUNT}/.trustee_done"
+CRYPTTAB_FILE="/sysroot/etc/crypttab"
+MAX_RETRY_ATTEMPTS=3
+RETRY_DELAY=5
+
+# temp configuration
+TEMP_DIR=$(mktemp -d /tmp/secure_attestation_XXXXXX)
+PASSPHRASE_FILE="${TEMP_DIR}/passphrase"
+OLD_PASS_FILE="${TEMP_DIR}/old_passphrase"
+
+# Make sure the safety of temp files
+umask 077
+# Clean function - make sure it is always executed
+cleanup() {
+    # Securely clean the sensitive data from memory
+    if [ -n "${passphrase}" ]; then
+        passphrase="xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx"
+    fi
+    
+    # umount the mountpoint
+    umount "${BOOT_MOUNT}" 2>/dev/null || true
+    
+    # Delete the temp files or directories
+    rm -rf "${TEMP_DIR}"
+    
+    info "*** ATTESTATION SERVICE COMPLETED ***"
+}
+trap cleanup EXIT
+
+fetch_passphrase() {
+    local attempts=$MAX_RETRY_ATTEMPTS
+    local count=0
+    local result=""
+    local status=0
+
+    while [ $count -lt $attempts ] && [ -z "$result" ]; do
+        info "Attempt $((count+1)): Fetching passphrase from ${KBS_URL}"
+	
+        if ! result=$(/usr/bin/trustee-attester --url "${KBS_URL}" get-resource --path default/rootdecrypt/key1 2>/dev/null); then
```

Comment:
This path isn't standard, it depends where the secret has been uploaded in the KBS. Is there a way to configure it? Or at least it will be better to use a variable to highlight that this is configurable